src/ng/filter/limitTo.js

```diff
@@ -108,19 +108,25 @@ function limitToFilter() {
     if (isNaN(limit)) return input;
 
     if (isNumber(input)) input = input.toString();
-    if (!isArray(input) && !isString(input)) return input;
+    if (!isArrayLike(input)) return input;
 
     begin = (!begin || isNaN(begin)) ? 0 : toInt(begin);
     begin = (begin < 0) ? Math.max(0, input.length + begin) : begin;
 
     if (limit >= 0) {
-      return input.slice(begin, begin + limit);
+      return sliceFn(input, begin, begin + limit);
     } else {
       if (begin === 0) {
-        return input.slice(limit, input.length);
+        return sliceFn(input, limit, input.length);
       } else {
-        return input.slice(Math.max(0, begin + limit), begin);
+        return sliceFn(input, Math.max(0, begin + limit), begin);
       }
     }
   };
 }
+
+function sliceFn(input, idx, length) {
+  if (isString(input)) return input.slice(idx, length);
```

Comment:
One alternative implementation I can do is instead of the `isString(input)` check, I can do
```js
return input.slice ? input.slice.call(input, idx, length) : slice.call(input, idx, length);
```
Thoughts?